src/nbl/video/ILogicalDevice.cpp

```diff
@@ -336,47 +337,42 @@ core::smart_refctd_ptr<IGPUShader> ILogicalDevice::createShader(const SShaderCre
             break;
     }
 
-    core::smart_refctd_ptr<const asset::ICPUShader> spirvShader;
-    if (creationParams.cpushader->getContentType()==asset::ICPUShader::E_CONTENT_TYPE::ECT_SPIRV)
-        spirvShader = core::smart_refctd_ptr<const asset::ICPUShader>(creationParams.cpushader);
-    else
-    {
-        auto compiler = m_compilerSet->getShaderCompiler(creationParams.cpushader->getContentType());
```

Comment:
The check if input is already in SPIR-V was useful, otherwise you'll get a nullptr crash as no compiler for the content type will be found.

Alternatively, you can do into the `else` case of the content type on line 370 and still run the SPIR-V optimizer and not the compiler set.